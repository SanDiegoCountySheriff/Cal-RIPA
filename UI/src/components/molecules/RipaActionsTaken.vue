<template>
  <div class="ripa-action-taken tw-pb-4">
    <ripa-form-header
      title="Actions Taken During Stop"
      required
      subtitle="ยง999.226(a)(12)"
      :on-open-statute="onOpenStatute"
    >
    </ripa-form-header>

    <v-container>
      <v-row no-gutters>
        <v-col cols="12" sm="12">
          <ripa-switch
            v-model="model.actionsTaken.anyActionsTaken"
            label="Any Actions Taken?"
            :disabled="isAnyActionsTakenDisabled1 || isAnyActionsTakenDisabled2"
            :max-width="200"
            @input="handleInput"
          ></ripa-switch>

          <template v-if="model.actionsTaken.anyActionsTaken">
            <ripa-check-group
              v-model="model.actionsTaken.actionsTakenDuringStop"
              :items="getActionsTakenGeneralItems"
              :rules="actionsTakenRules"
              @input="handleInput"
            >
            </ripa-check-group>

            <ripa-subheader text="Search"></ripa-subheader>

            <ripa-check-group
              v-model="model.actionsTaken.actionsTakenDuringStop"
              :items="getActionsTakenSearchItems"
              :rules="actionsTakenRules"
              @input="handleInput"
            >
            </ripa-check-group>

            <template v-if="wasAskedForConsentToSearchPerson">
              <ripa-switch
                v-model="model.actionsTaken.personSearchConsentGiven"
                label="Person Search Consent Given"
                :max-width="300"
                :rules="personPropertySearchConsentGivenRules"
                @input="handleInput"
              ></ripa-switch>
            </template>

            <template v-if="wasAskedForConsentToSearchProperty">
              <ripa-switch
                v-model="model.actionsTaken.propertySearchConsentGiven"
                label="Property Search Consent Given"
                :max-width="300"
                :rules="personPropertySearchConsentGivenRules"
                @input="handleInput"
              ></ripa-switch>
            </template>

            <template v-if="wasSearchOfPersonOrPropertyConducted">
              <ripa-form-subheader
                title="Basis for Search"
                required
                subtitle="ยง999.226(a)(12)(B)"
                :on-open-statute="onOpenStatute"
              ></ripa-form-subheader>

              <ripa-check-group
                v-model="model.actionsTaken.basisForSearch"
                :items="getBasisForSearchItems"
                :rules="basisForSearchRules"
                @input="handleInput"
              >
              </ripa-check-group>

              <template v-if="isBasisForSearchExplanationVisible">
                <template v-if="model.actionsTaken.basisForSearchPiiFound">
                  <ripa-alert alert-outlined alert-type="warning">
                    The explanation contains personally identifying information.
                    Please remove if possible.
                  </ripa-alert>
                </template>

                <ripa-text-input
                  v-model="model.actionsTaken.basisForSearchExplanation"
                  hint="Important: Do not include personally identifying information, such as names, DOBs, addresses, ID numbers, etc."
                  persistent-hint
                  label="Brief Explanation"
                  :loading="loadingPii"
                  :rules="explanationRules"
                  @input="handleInput($event), handlePiiCheck($event)"
                ></ripa-text-input>
              </template>
            </template>

            <ripa-subheader text="Seizure"></ripa-subheader>

            <ripa-switch
              v-model="model.actionsTaken.propertyWasSeized"
              label="Property was Seized"
              :max-width="200"
              @input="handleInput"
            ></ripa-switch>

            <template v-if="model.actionsTaken.propertyWasSeized">
              <ripa-form-subheader
                title="Basis for Property Seizure"
                required
                subtitle="ยง999.226(a)(12)(D)(1)"
                :on-open-statute="onOpenStatute"
              ></ripa-form-subheader>

              <ripa-check-group
                v-model="model.actionsTaken.basisForPropertySeizure"
                :items="getBasisForPropertySeizureItems"
                :rules="basisForPropertySeizureRules"
                @input="handleInput"
              >
              </ripa-check-group>

              <ripa-form-subheader
                title="Types of Property Seized"
                required
                subtitle="ยง999.226(a)(12)(D)(2)"
                :on-open-statute="onOpenStatute"
              ></ripa-form-subheader>

              <ripa-check-group
                v-model="model.actionsTaken.typeOfPropertySeized"
                :items="propertySeizedTypeItems"
                :rules="typeOfPropertySeizedRules"
                @input="handleInput"
              >
              </ripa-check-group>
            </template>
          </template>
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script>
import RipaAlert from '@/components/atoms/RipaAlert'
import RipaFormHeader from '@/components/molecules/RipaFormHeader'
import RipaModelMixin from '@/components/mixins/RipaModelMixin'
import RipaCheckGroup from '@/components/atoms/RipaCheckGroup'
import RipaFormSubheader from '@/components/molecules/RipaFormSubheader'
import RipaSubheader from '@/components/atoms/RipaSubheader'
import RipaSwitch from '@/components/atoms/RipaSwitch'
import RipaTextInput from '@/components/atoms/RipaTextInput'
import {
  ACTIONS_TAKEN,
  BASIS_FOR_SEARCH,
  BASIS_FOR_PROPERTY_SEIZURE,
  SEIZED_PROPERTY_TYPES,
} from '@/constants/form'

export default {
  name: 'ripa-action-taken',

  mixins: [RipaModelMixin],

  components: {
    RipaAlert,
    RipaFormHeader,
    RipaCheckGroup,
    RipaFormSubheader,
    RipaSubheader,
    RipaSwitch,
    RipaTextInput,
  },

  data() {
    return {
      explanationRules: [
        v => (v || '').length > 0 || 'Explanation is required',
        v => (v || '').length <= 250 || 'Max 250 characters',
        v => (v || '').length >= 5 || 'Min 5 characters',
      ],
      actionsTakenItems: ACTIONS_TAKEN,
      basisForSearchItems: BASIS_FOR_SEARCH,
      basisForPropertySeizureItems: BASIS_FOR_PROPERTY_SEIZURE,
      isAnyActionsTakenDisabled1: false,
      isAnyActionsTakenDisabled2: false,
      propertySeizedTypeItems: SEIZED_PROPERTY_TYPES,
      viewModel: this.syncModel(this.value),
    }
  },

  created() {
    if (this.viewModel.actionsTaken.basisForSearchExplanation) {
      this.handlePiiCheck(this.viewModel.actionsTaken.basisForSearchExplanation)
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
    },

    actionsTakenRules() {
      const checked = this.viewModel.actionsTaken.anyActionsTaken
      const options = this.viewModel.actionsTaken.actionsTakenDuringStop.filter(
        item => item !== 21,
      )

      return [
        (checked && options.length > 0) ||
          'At least one action taken is required',
      ]
    },

    basisForSearchRules() {
      const consentGiven = this.wasAskedForConsent
      const searchConducted = this.wasSearchOfPersonOrPropertyConducted
      const options = this.viewModel.actionsTaken.basisForSearch

      return [
        (searchConducted && options.length > 0) ||
          'At least one basis for search is required',
        !consentGiven ||
          (consentGiven &&
            this.viewModel.actionsTaken.basisForSearch !== null &&
            this.viewModel.actionsTaken.basisForSearch.length > 0 &&
            this.viewModel.actionsTaken.basisForSearch.includes(1)) ||
          'Consent given must be selected if person or property consent was given.',
      ]
    },

    basisForPropertySeizureRules() {
      const checked = this.viewModel.actionsTaken.propertyWasSeized
      const options = this.viewModel.actionsTaken.basisForPropertySeizure

      return [
        (checked && options.length > 0) ||
          'At least one basis for property seizure is required',
      ]
    },

    personPropertySearchConsentGivenRules() {
      const checked =
        this.viewModel.actionsTaken.personSearchConsentGiven ||
        this.viewModel.actionsTaken.propertySearchConsentGiven
      const basisForSearch = this.viewModel.actionsTaken?.basisForSearch || []
      const consentGiven = basisForSearch.includes(1)

      if (!consentGiven) {
        return []
      }

      return [
        (checked && consentGiven) ||
          '"Basis for Search" indicates "Consent Given" but Person search consent or Property search consent has not been selected',
      ]
    },

    typeOfPropertySeizedRules() {
      const checked = this.viewModel.actionsTaken.propertyWasSeized
      const options = this.viewModel.actionsTaken.typeOfPropertySeized

      return [
        (checked && options.length > 0) ||
          'At least one type of property seized is required',
      ]
    },

    getActionsTakenGeneralItems() {
      const filteredItems = this.actionsTakenItems.filter(
        item => ![17, 18, 19, 20, 21].includes(item.value),
      )

      if (!this.viewModel.person.isStudent) {
        return filteredItems.filter(item => item.value !== 23)
      }

      return filteredItems
    },

    getActionsTakenSearchItems() {
      return this.actionsTakenItems
        .filter(item => [17, 18, 19, 20].includes(item.value))
        .map(item => {
          return {
            ...item,
            disabled:
              (this.isAnyActionsTakenDisabled1 && item.value === 18) ||
              (this.isAnyActionsTakenDisabled2 && item.value === 20),
          }
        })
    },

    getBasisForSearchItems() {
      const actionsTaken =
        this.viewModel.actionsTaken?.actionsTakenDuringStop || []
      let filteredItems = this.basisForSearchItems

      if (!this.viewModel.person.isStudent) {
        filteredItems = filteredItems.filter(item => item.value !== 13)
      }

      if (actionsTaken.includes(20)) {
        return filteredItems
      }

      return filteredItems.filter(item => item.value !== 12)
    },

    getBasisForPropertySeizureItems() {
      let filteredItems = this.basisForPropertySeizureItems

      if (!this.viewModel.person.isStudent) {
        filteredItems = filteredItems.filter(item => item.value !== 6)
      }

      return filteredItems
    },

    wasSearchOfPersonOrPropertyConducted() {
      const actionsTaken =
        this.viewModel.actionsTaken?.actionsTakenDuringStop || []
      return actionsTaken.includes(18) || actionsTaken.includes(20)
    },

    wasAskedForConsent() {
      return (
        this.viewModel.actionsTaken.personSearchConsentGiven ||
        this.viewModel.actionsTaken.propertySearchConsentGiven
      )
    },

    wasAskedForConsentToSearchPerson() {
      return this.viewModel.actionsTaken.actionsTakenDuringStop.includes(17)
    },

    wasAskedForConsentToSearchProperty() {
      return this.viewModel.actionsTaken.actionsTakenDuringStop.includes(19)
    },

    isBasisForSearchExplanationVisible() {
      if (this.viewModel.actionsTaken.basisForSearch.length === 0) {
        return false
      }

      if (
        this.viewModel.actionsTaken.basisForSearch.length === 1 &&
        this.viewModel.actionsTaken.basisForSearch.includes(4)
      ) {
        return false
      }

      return this.wasSearchOfPersonOrPropertyConducted
    },
  },

  methods: {
    handleInput() {
      this.updateModel()
      this.$emit('input', this.viewModel)
    },

    handlePiiCheck(textValue) {
      this.$emit('pii-check', { source: 'search', value: textValue })
    },
  },

  mounted() {
    this.updateModel()
  },

  watch: {
    value(newVal) {
      this.viewModel = this.syncModel(newVal)
      this.updateModel()
    },
  },

  props: {
    value: {
      type: Object,
      default: () => {},
    },
    loadingPii: {
      type: Boolean,
      default: false,
    },
  },
}
</script>
