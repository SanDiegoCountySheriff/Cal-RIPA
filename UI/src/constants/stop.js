export const SUBMISSION_STATUSES = [
  {
    text: 'Failed',
    value: 'Failed',
    editable: true,
  },
  {
    text: 'Submitted',
    value: 'Submitted',
    editable: false,
  },
  {
    text: 'Unsubmitted',
    value: 'Unsubmitted',
    editable: true,
  },
  {
    text: 'Resubmitted',
    value: 'Resubmitted',
    editable: false,
  },
]
