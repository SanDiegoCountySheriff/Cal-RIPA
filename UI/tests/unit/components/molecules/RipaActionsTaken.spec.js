import RipaActionsTaken from '@/components/molecules/RipaActionsTaken'
import RipaModelMixin from '@/components/mixins/RipaModelMixin.vue'
import { defaultStop } from '@/utilities/stop.js'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Actions Taken', () => {
  let vuetify
  let wrapper
  let stop

  beforeEach(() => {
    vuetify = new Vuetify()
    stop = defaultStop()
  })

  const factory = propsData => {
    return shallowMount(RipaActionsTaken, {
      vuetify,
      propsData: {
        ...propsData,
      },
      mixins: [RipaModelMixin],
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaActionsTaken, { vuetify, propsData: { value: stop } })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should validate explanation rules', () => {
    wrapper = factory({ value: stop })
    const nullTestCase = {
      testValue: null,
      firstTest: 'Explanation is required',
      secondTest: true,
      thirdTest: 'Min 5 characters',
    }
    const shortTestCase = {
      testValue: 'test',
      firstTest: true,
      secondTest: true,
      thirdTest: 'Min 5 characters',
    }
    const longTestCase = {
      testValue: '',
      firstTest: true,
      secondTest: 'Max 250 characters',
      thirdTest: true,
    }
    const validTestCase = {
      testValue: 'Testing',
      firstTest: true,
      secondTest: true,
      thirdTest: true,
    }

    for (let i = 0; i <= 255; i++) {
      longTestCase.testValue += 'a'
    }

    const testCases = [nullTestCase, shortTestCase, longTestCase, validTestCase]

    for (const testCase of testCases) {
      expect(wrapper.vm.explanationRules[0](testCase.testValue)).toEqual(
        testCase.firstTest,
      )
      expect(wrapper.vm.explanationRules[1](testCase.testValue)).toEqual(
        testCase.secondTest,
      )
      expect(wrapper.vm.explanationRules[2](testCase.testValue)).toEqual(
        testCase.thirdTest,
      )
    }
  })

  it('should handle pii check when created', () => {
    stop.actionsTaken.basisForSearchExplanation = 'Testing'
    wrapper = factory({ value: stop })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should validate actions taken rules', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.vm.actionsTakenRules).toEqual([
      'At least one action taken is required',
    ])

    wrapper.vm.viewModel.actionsTaken.anyActionsTaken = true
    wrapper.vm.viewModel.actionsTaken.nonForceActionsTakenDuringStop = [1]

    expect(wrapper.vm.actionsTakenRules).toEqual([true])
  })

  it('should validate basis for search rules', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.vm.basisForSearchRules).toEqual([
      'At least one basis for search is required',
      true,
    ])

    wrapper.vm.viewModel.actionsTaken.personSearchConsentGiven = true
    wrapper.vm.viewModel.actionsTaken.nonForceActionsTakenDuringStop = [10]
    wrapper.vm.viewModel.actionsTaken.basisForSearch = [2]

    expect(wrapper.vm.basisForSearchRules).toEqual([
      true,
      'Consent given must be selected if person or property consent was given.',
    ])

    wrapper.vm.viewModel.actionsTaken.personSearchConsentGiven = true
    wrapper.vm.viewModel.actionsTaken.nonForceActionsTakenDuringStop = [10]
    wrapper.vm.viewModel.actionsTaken.basisForSearch = [1]

    expect(wrapper.vm.basisForSearchRules).toEqual([true, true])
  })

  it('should validate basis for property seizure rules', () => {
    wrapper = factory({ value: stop })

    wrapper.vm.viewModel.actionsTaken.propertyWasSeized = true
    wrapper.vm.viewModel.actionsTaken.typeOfPropertySeized = []
    wrapper.vm.viewModel.actionsTaken.basisForPropertySeizure = []

    expect(wrapper.vm.basisForPropertySeizureRules).toEqual([
      'At least one basis for property seizure is required',
    ])

    wrapper.vm.viewModel.actionsTaken.basisForPropertySeizure = [1]

    expect(wrapper.vm.basisForPropertySeizureRules).toEqual([true])
  })

  it('should validate person property search consent given rules', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.vm.personPropertySearchConsentGivenRules).toEqual([])

    wrapper.vm.viewModel.actionsTaken.basisForSearch = [1]

    expect(wrapper.vm.personPropertySearchConsentGivenRules).toEqual([
      '"Basis for Search" indicates "Consent Given" but Person search consent or Property search consent has not been selected',
    ])

    wrapper.vm.viewModel.actionsTaken.personSearchConsentGiven = true

    expect(wrapper.vm.personPropertySearchConsentGivenRules).toEqual([true])
  })

  it('should validate type of property seized rules', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.vm.typeOfPropertySeizedRules).toEqual([
      'At least one type of property seized is required',
    ])

    wrapper.vm.viewModel.actionsTaken.propertyWasSeized = true
    wrapper.vm.viewModel.actionsTaken.typeOfPropertySeized = []
    wrapper.vm.viewModel.actionsTaken.basisForPropertySeizure = []

    expect(wrapper.vm.typeOfPropertySeizedRules).toEqual([
      'At least one type of property seized is required',
    ])

    wrapper.vm.viewModel.actionsTaken.typeOfPropertySeized = [1]

    expect(wrapper.vm.typeOfPropertySeizedRules).toEqual([true])
  })

  it('should get actions taken general items', () => {
    wrapper = factory({ value: stop })
    wrapper.vm.actionsTakenItems = [
      { value: 1 },
      { value: 2 },
      { value: 8 },
      { value: 10 },
      { value: 17 },
    ]

    wrapper.vm.viewModel.person.isStudent = true
    expect(wrapper.vm.getActionsTakenGeneralItems).toEqual([
      { value: 1 },
      { value: 2 },
      { value: 17 },
    ])

    wrapper.vm.viewModel.person.isStudent = false

    expect(wrapper.vm.getActionsTakenGeneralItems).toEqual([
      { value: 1 },
      { value: 2 },
    ])
  })

  it('should get actions taken search items', () => {
    wrapper = factory({ value: stop })
    wrapper.vm.actionsTakenItems = [
      { value: 1 },
      { value: 2 },
      { value: 8 },
      { value: 10 },
      { value: 11 },
      { value: 12 },
      { value: 17 },
    ]

    expect(wrapper.vm.getActionsTakenSearchItems).toEqual([
      { value: 8, disabled: false },
      { value: 10, disabled: false },
      { value: 11, disabled: false },
      { value: 12, disabled: false },
    ])

    wrapper.vm.isAnyActionsTakenDisabled1 = true
    wrapper.vm.isAnyActionsTakenDisabled2 = true

    expect(wrapper.vm.getActionsTakenSearchItems).toEqual([
      { value: 8, disabled: false },
      { value: 10, disabled: true },
      { value: 11, disabled: false },
      { value: 12, disabled: true },
    ])
  })

  it('should get basis for search items', () => {
    wrapper = factory({ value: stop })
    wrapper.vm.basisForSearchItems = [
      { value: 1 },
      { value: 2 },
      { value: 12 },
      { value: 13 },
    ]

    wrapper.vm.viewModel.actionsTaken.nonForceActionsTakenDuringStop = null

    expect(wrapper.vm.getBasisForSearchItems).toEqual([
      { value: 1 },
      { value: 2 },
    ])

    wrapper.vm.viewModel.actionsTaken.nonForceActionsTakenDuringStop = [12]
    wrapper.vm.viewModel.actionsTaken.basisForSearch = []

    expect(wrapper.vm.getBasisForSearchItems).toEqual([
      { value: 1 },
      { value: 2 },
      { value: 12 },
    ])

    wrapper.vm.viewModel.person.isStudent = true

    expect(wrapper.vm.getBasisForSearchItems).toEqual([
      { value: 1 },
      { value: 2 },
      { value: 12 },
      { value: 13 },
    ])
  })

  it('should get basis for property seizure items', () => {
    wrapper = factory({ value: stop })
    wrapper.vm.basisForPropertySeizureItems = [
      { value: 1 },
      { value: 2 },
      { value: 6 },
    ]

    expect(wrapper.vm.getBasisForPropertySeizureItems).toEqual([
      { value: 1 },
      { value: 2 },
    ])

    wrapper.vm.viewModel.person.isStudent = true

    expect(wrapper.vm.getBasisForPropertySeizureItems).toEqual([
      { value: 1 },
      { value: 2 },
      { value: 6 },
    ])
  })

  it('should validate was search of person or property conducted', () => {
    wrapper = factory({ value: stop })

    wrapper.vm.viewModel.actionsTaken.nonForceActionsTakenDuringStop = null

    expect(wrapper.vm.wasSearchOfPersonOrPropertyConducted).toBeFalsy()

    wrapper.vm.viewModel.actionsTaken.nonForceActionsTakenDuringStop = [10]
    wrapper.vm.viewModel.actionsTaken.basisForSearch = []

    expect(wrapper.vm.wasSearchOfPersonOrPropertyConducted).toBeTruthy()
  })

  it('should validate was asked for consent', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.vm.wasAskedForConsent).toBeFalsy()

    wrapper.vm.viewModel.actionsTaken.personSearchConsentGiven = true

    expect(wrapper.vm.wasAskedForConsent).toBeTruthy()
  })

  it('should validate was asked for consent to search person', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.vm.wasAskedForConsentToSearchPerson).toBeFalsy()

    wrapper.vm.viewModel.actionsTaken.nonForceActionsTakenDuringStop = [8]

    expect(wrapper.vm.wasAskedForConsentToSearchPerson).toBeTruthy()
  })

  it('should validate was asked for consent to search property', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.vm.wasAskedForConsentToSearchProperty).toBeFalsy()

    wrapper.vm.viewModel.actionsTaken.nonForceActionsTakenDuringStop = [11]

    expect(wrapper.vm.wasAskedForConsentToSearchProperty).toBeTruthy()
  })

  it('should display basis for search explanation when valid', () => {
    wrapper = factory({ value: stop })

    wrapper.vm.viewModel.actionsTaken.basisForSearch = []

    expect(wrapper.vm.isBasisForSearchExplanationVisible).toBeFalsy()

    wrapper.vm.viewModel.actionsTaken.basisForSearch = [4]

    expect(wrapper.vm.isBasisForSearchExplanationVisible).toBeFalsy()

    wrapper.vm.viewModel.actionsTaken.basisForSearch = [3]
    wrapper.vm.viewModel.actionsTaken.nonForceActionsTakenDuringStop = [10]

    expect(wrapper.vm.isBasisForSearchExplanationVisible).toBeTruthy()
  })

  it('should handle input', () => {
    wrapper = factory({ value: stop })

    wrapper.vm.handleInput()

    expect(wrapper.emitted('input')).toBeTruthy()
  })

  it('should handle check pii', () => {
    wrapper = factory({ value: stop })

    wrapper.vm.handlePiiCheck('test')

    expect(wrapper.emitted('pii-check')).toBeTruthy()
    expect(wrapper.emitted('pii-check')[0][0].value).toEqual('test')
  })

  it('should watch value', async () => {
    wrapper = factory({ value: stop })

    const updatedStop = defaultStop()
    updatedStop.id = 1

    wrapper.setProps({ value: updatedStop })
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.viewModel.id).toEqual(1)
  })
})
