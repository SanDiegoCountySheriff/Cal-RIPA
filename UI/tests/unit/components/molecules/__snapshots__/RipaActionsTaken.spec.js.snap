// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Actions Taken should handle pii check when created 1`] = `
<div class="ripa-action-taken tw-pb-4">
  <ripa-form-header-stub title="Actions Taken During Stop" required="true" subtitle="ยง999.226(a)(12)" onopenstatute="[Function]"></ripa-form-header-stub>
  <v-container-stub tag="div">
    <v-row-stub tag="div" nogutters="true">
      <v-col-stub cols="12" sm="12" tag="div">
        <ripa-switch-stub value="true" label="Any Actions Taken?" hint="" maxwidth="200" rules=""></ripa-switch-stub>
        <ripa-check-group-stub value="" items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" rules="At least one action taken is required"></ripa-check-group-stub>
        <ripa-subheader-stub text="Search"></ripa-subheader-stub>
        <ripa-check-group-stub value="" items="[object Object],[object Object],[object Object],[object Object]" rules="At least one action taken is required"></ripa-check-group-stub>
        <!---->
        <!---->
        <!---->
        <ripa-subheader-stub text="Seizure"></ripa-subheader-stub>
        <ripa-switch-stub label="Property was Seized" hint="" maxwidth="200" rules=""></ripa-switch-stub>
        <!---->
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`Ripa Actions Taken should match snapshot 1`] = `
<div
  class="ripa-action-taken tw-pb-4"
>
  <ripa-form-header-stub
    onopenstatute="[Function]"
    required="true"
    subtitle="ยง999.226(a)(12)"
    title="Actions Taken During Stop"
  />
   
  <v-container-stub
    tag="div"
  >
    <v-row-stub
      nogutters="true"
      tag="div"
    >
      <v-col-stub
        cols="12"
        sm="12"
        tag="div"
      >
        <ripa-switch-stub
          hint=""
          label="Any Actions Taken?"
          maxwidth="200"
          rules=""
          value="true"
        />
         
        <ripa-check-group-stub
          items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
          rules="At least one action taken is required"
          value=""
        />
         
        <ripa-subheader-stub
          text="Search"
        />
         
        <ripa-check-group-stub
          items="[object Object],[object Object],[object Object],[object Object]"
          rules="At least one action taken is required"
          value=""
        />
         
        <!---->
         
        <!---->
         
        <!---->
         
        <ripa-subheader-stub
          text="Seizure"
        />
         
        <ripa-switch-stub
          hint=""
          label="Property was Seized"
          maxwidth="200"
          rules=""
        />
         
        <!---->
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;
